packages/components/src/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,211 @@
+import clsx from "clsx";
+import React from "react";
+import { useUID } from "react-uid";
+import styles from "./Checkbox.module.css";
+
+// Allow all native HTML input attributes,
+// except for ones where we override type or strictness
+type CheckboxHTMLElementProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  "checked" | "id" | "size"
+>;
+
+type InputProps = CheckboxHTMLElementProps & {
+  /**
+   * Whether checkbox is checked. Defaults to false.
+   * "indeterminate" can be used when a checkbox visually represents
+   * a list of checkboxes that is "partially" checked.
+   */
+  checked?: boolean | "indeterminate";
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+  /**
+   * Checkbox ID. Used to connect the input with a label for accessibility purposes.
+   */
+  id: string;
+};
+
+type LabelProps = {
+  /**
+   * Text to render in label.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classnames passed in for styling.
+   */
+  className?: string;
+  /**
+   * ID of input that label is associated with.
+   */
+  htmlFor: string;
+  /**
+   * Size of the checkbox label.
+   */
+  size?: "small" | "medium";
+};
+
+// id is required in InputProps but optional in CheckboxProps, so we
+// first remove `id` from InputProps before intersecting.
+export type CheckboxProps = Omit<InputProps, "id"> & {
+  /**
+   * HTML id attribute. If not passed, this component
+   * will generate an id to use for accessibility.
+   */
+  id?: string;
+  /**
+   * Visible text label for the checkbox.
+   */
+  label?: React.ReactNode;
+  /**
+   * Size of the checkbox label.
+   */
+  size?: LabelProps["size"];
+};
+
+interface SvgStyle extends React.CSSProperties {
+  "--checkbox-svg-size": string;
+}
```

Comment:
This reminds me, I wonder if we should export a type definition with all css variables we use. That way LP can consumers can supply values for these in style attributes with max autocompletion and type safety.

Reply:
ooh I like it. tracked: https://app.shortcut.com/czi-edu/story/180821/export-type-definition-from-eds-with-all-css-variables-used